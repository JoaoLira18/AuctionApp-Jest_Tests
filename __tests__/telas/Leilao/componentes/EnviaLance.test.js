import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react-native'
import { EnviaLance } from '../../../../src/telas/Leilao/componentes/EnviaLance.js'
import { ENVIADO, NAO_ENVIADO } from '../../../../src/negocio/constantes/estadosLance';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

describe('telas/Leilao/componentes/EnviaLance', () => {

    it('must send a bid when press the button', async () => {

        const enviaLance = jest.fn(() => new Promise(resolve => resolve(ENVIADO)))
        const {
            getByText,
            getByPlaceholderText,
            getByAccessibilityHint,
        } = render(
            <EnviaLance
                enviaLance={enviaLance}
                cor="blue"
            />
        )

        const input = getByPlaceholderText("R$");
        const button = getByAccessibilityHint("Enviar Lance");

        fireEvent.changeText(input, "10");
        fireEvent.press(button);

        expect(enviaLance).toHaveBeenCalledWith("10");
        await waitFor(() => {
            expect(getByText(ENVIADO)).toBeTruthy()
        })

        expect(() => getByText(NAO_ENVIADO)).toThrow();
    })
})

describe("testing enviaLance snapshot ", () => {

    it("must render correctly", () => {
        const enviaLance = jest.fn(() => new Promise(resolve => resolve(ENVIADO)))
        const { toJSON } = render(
            <EnviaLance
                enviaLance={enviaLance}
                cor="blue"
            />
        )

        expect(toJSON()).toMatchSnapshot()
    })
})