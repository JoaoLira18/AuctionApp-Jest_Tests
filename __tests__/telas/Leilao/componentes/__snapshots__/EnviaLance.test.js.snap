// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing enviaLance snapshot  must render correctly 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "left": 0,
      "padding": 8,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <TextInput
    editable={true}
    keyboardType="decimal-pad"
    onChangeText={[Function]}
    placeholder="R$"
    style={
      Object {
        "borderColor": "#4A4A4A",
        "borderRadius": 16,
        "borderWidth": 1,
        "color": "#4A4A4A",
        "flex": 1,
        "fontSize": 16,
        "height": 53,
        "margin": 8,
        "padding": 16,
        "textAlign": "center",
      }
    }
    value=""
  />
  <View
    accessibilityHint="Enviar Lance"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
        "borderRadius": 16,
        "margin": 8,
        "opacity": 1,
        "padding": 16,
      }
    }
  />
</View>
`;
